controllers/storageclaim_controller.go

```diff
@@ -501,64 +544,111 @@ func (r *StorageClaimReconciler) getCephRBDStorageClass(data map[string]string)
-		return fmt.Errorf("failed to own storageclass: %v", err)
+	existing.SetName(r.storageClaim.Name)
+
+	if err := r.own(class); err != nil {
+		return fmt.Errorf("failed to own %s: %v", classKind, err)
 	}
 
 	if err := r.get(existing); err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("failed to get StorageClass: %v", err)
+		return fmt.Errorf("failed to get %s: %v", classKind, err)
 	}
 
-	// If present then compare the existing StorageClass with the received StorageClass, and only proceed if they differ.
-	if reflect.DeepEqual(existing.Parameters, storageClass.Parameters) {
+	// If present then compare the existing Storage(Volume*Snapshot)Class parameters and labels with
+	// the received Storage(Volume*Snapshot)Class parameters and labels, and only proceed if they differ.
+	labelsEqual := reflect.DeepEqual(existing.GetLabels(), class.GetLabels())
+	parametersEqual := reflect.DeepEqual(existingParameters, classParameters)
+	annotationsEqual := reflect.DeepEqual(existing.GetAnnotations(), class.GetAnnotations())
+	if labelsEqual && parametersEqual && annotationsEqual {
 		return nil
 	}
 
-	// StorageClass already exists, but parameters have changed. Delete the existing StorageClass and create a new one.
-	if existing.UID != "" {
-
-		// Since we have to update the existing StorageClass, so we will delete the existing StorageClass and create a new one.
-		r.log.Info("StorageClass needs to be updated, deleting it.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
-
-		// Delete the StorageClass.
-		err := r.delete(existing)
-		if err != nil {
-			r.log.Error(err, "Failed to delete StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
-			return fmt.Errorf("failed to delete StorageClass: %v", err)
+	// Storage(Volume*Snapshot)Class already exists, but parameters have changed. Delete the existing
+	// Storage(Volume*Snapshot)Class and create a new one.
+	if existing.GetUID() != "" {
+		// Since we have to update the existing Storage(Volume*Snapshot)Class, so we will delete the existing
+		// Storage(Volume*Snapshot)Class and create a new one.
+		r.log.Info(classKind+" needs to be updated, deleting it.", "Name", types.NamespacedName{
+			Namespace: class.GetNamespace(),
+			Name:      existing.GetName(),
+		}.String())
+
+		// Delete the Storage(Volume*Snapshot)Class.
+		if err := r.delete(existing); err != nil {
+			r.log.Error(err, "Failed to delete %s.", "Name", types.NamespacedName{
+				Namespace: class.GetNamespace(),
+				Name:      existing.GetName(),
+			}.String())
+			return fmt.Errorf("failed to delete "+classKind+": %v", err)
 		}
 	}
-	r.log.Info("Creating StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
-	err := r.Client.Create(r.ctx, storageClass)
-	if err != nil {
-		return fmt.Errorf("failed to create StorageClass: %v", err)
+	r.log.Info("Creating "+classKind+".", "Name", types.NamespacedName{
+		Namespace: class.GetNamespace(),
```

Comment:
namespace is not required as its clusterScoped resouces

Reply:
And how about for `StoageClass`? Seems like it did have its namespace mentioned in the logs:
```go
r.log.Info("StorageClass needs to be updated, deleting it.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
```

Reply:
it looks like that is also not required but it was present